publish.go

```diff
@@ -69,5 +67,7 @@ func (h *Hub) PublishHandler(w http.ResponseWriter, r *http.Request) {
 	io.WriteString(w, u.ID)
 	h.logger.Info("Update published", zap.Object("update", u), zap.String("remote_addr", r.RemoteAddr))
 
-	h.metrics.NewUpdate(u)
+	if h.metrics != nil {
```

Comment:
I think having a `noopMetrics` implementation and falling back to it in the constructor when no metrics is set using `WithMetrics` would be slightly better than checking `h.metrics` every time.

Reply:
Sure, it was a remainder of the last iteration.